packages/camera/camera_android/android/build.gradle

```diff
@@ -27,7 +27,7 @@ project.getTasks().withType(JavaCompile){
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 31
+    compileSdkVersion 33
```

Comment:
Do we really want to update the plugins themselves, rather than just the example apps?

Because of [this code](https://github.com/flutter/flutter/blob/6205c110d64351610219d48b52a1a1b9e0140dcd/packages/flutter_tools/gradle/flutter.gradle#L452-L485), this will create errors in all end-user's projects if they aren't using 33, even though nothing in the plugin actually requires 33 at this point.

I thought the behavior we wanted was that our *apps* should always be using the latest SDK, but plugins themselves should only be using the `compileSdkVersion` that they actually require for their code.

Reply:
Updating apps but not plugins is a reasonable decision and if that is our policy I will modify this PR to follow it. 

Here is the counter argument. 
* Compile sdk version changes have no runtime behavior differences. 
* It is generally considered best practice to have the highest stable compileSdkVersion. 
* Using a higher compileSdkVersion is strictly better (IMO), you get better lints/warnings etc. 
* Plugins are versioned so if developers lock the version they are safe. Maybe there is an argument that since this might break compile so I should change the version code to bump minor instead of patch. 
* CompileSdkVersion has been available for many months and google is on the verge of making compileSdkVersion 34 available. 
* I think our CI machines need the android sdk for every compileSdkVersion we use. 
* *Probably the weakest argument* but one I wanted to include, Our own devs can be confused by this number and why it might be different and I want to set the example that we should always use the highest stable version. 

I think policy that takes into account the above while also realizing that more people use flutter and flutter plugins than those that work on flutter (my inner Hixie voice) and also that I want to keep us up to date would be... 
1. Apps use highest available stable compileSdkVersion as defined in flutter.compileSdkVersion
1.  Plugins update minor version number X (1-4) months after a new stable compileSdkVersion is available 

Open to anyone's thoughts on this.

Reply:
If you like that suggestion let me know and I will modify the version numbers. If not then I will revert the changes to this and other plugins.

Reply:
> Maybe there is an argument that since this might break compile so I should change the version code to bump minor instead of patch.

This wouldn't really change anything for clients; the standard way of setting versions in pubspec.yaml (what `pub add` does by default, what all our docs do, etc.) is to use a `^` range, which accepts anything up to the next major version.

> * CompileSdkVersion has been available for many months and google is on the verge of making compileSdkVersion 34 available.

Well, the Flutter-SDK-provided version only changed to 33 in 3.7, which has only been out for a month.

In practice, only developers who either:
- originally made their app in the last ~year (Flutter 2.8 was when the template started using a variable, and that was released in Dec 2021) and updated Flutter itself to the last version in the last month, or
- manually manage their SDK version on a regular basis

will already be using 33. Basically everyone else (given the usage levels of plugins like path_provider and shared_preferences) will have to manually change their project the next time they update their dependencies if we land this as-is.

(The "made their app in the last year" requirement would be eliminated if we have the tool automigrate old hard-coded SDKs to the variable, as Jenn suggested elsewhere, but that would have to be done and then roll out to stable first.)

> I think policy that takes into account the above while also realizing that more people use flutter and flutter plugins than those that work on flutter (my inner Hixie voice) and also that I want to keep us up to date would be...
> 
> 1. Apps use highest available stable compileSdkVersion as defined in flutter.compileSdkVersion
> 2. Plugins update minor version number X (1-4) months after a new stable compileSdkVersion is available
> 
> Open to anyone's thoughts on this.

I'm fine with whatever you want to do here—we do have a clear build log message in place instructing people about what to do, which makes it much less disruptive—I just wanted to flag it to make sure it's a decision we're making deliberately since it's a change from what we've done historically AFAIK.

If you go with a policy like that one, I would suggest pinning part 2 to the stable Flutter release where we rev the template-referenced `compileSdkVersion` rather than the SDK release itself, since that's what I would expect most people to be using. Either "X months" or "after the *next* stable release" (which would generally map to about 3 months) could be reasonable options.

Whatever policy you decide on, ideally we should encode it in the repo tooling and CI so that it's not just relying on people remembering to update. If you file an issue I can add it to my current list of tool improvement work.

Reply:
@camsim99 @bparrishMines @gmackall @tarrinneal What do yall think about bumping the version used in the plugins even if the plugins do not use features from the newer compile sdk?

Reply:
If it doesn't break usage on older devices, I think it's reasonable to bump them all. No one likes churn, but it seems like it shouldn't be significantly difficult for people to make this change.

Reply:
Im in support of all plugins updating the hard coded number even if it isn't required. Requiring apps to update their `compileSdkVersion` to use the latest version of a dependency is just an inherent process for Android development and it seems we were able to make it pretty trivial to handle. If I was developing a pure Android app and wanted to update my `webkit` from `1.5.0` to `1.6.0`, I would have to update my `compileSdkVersion` too. I lean towards pinning it to the version set in flutter_tools though.

The issue I have with `flutter.compileSdkVersion` is that the packages repo develops on the flutter `main` branch, but I would imagine most apps are developed on the flutter `stable` branch. This could lead to a scenario were we update a plugin dependency that requires a higher `compileSdkVersion` than what is used by the Flutter stable branch. But if using the highest `compileSdkVersion` is a strict improvement, then this may not be much of an issue.

Reply:
Thanks to everyone for their feedback. I will leave these upgraded and we will try to keep the plugins synced to the version used in flutter.compileSdkVersion.